ts/main.ts

```diff
@@ -0,0 +1,201 @@
+// API Key Variables
+const key1 = 'sk-proj-7vyDO4PFulA9RzJM_KxUIZtVTUOmdlNR7Oy8D';
+const key2 = 'q1a8rQtNWWnRJ3rRtrmGJu808dnJveOjer0dVT3BlbkFJI';
+const key3 = 'uqxWfNJ9rai7axHkcNhLUvJVSW1f-pksYl4jIt8Dvq9eeFM';
+const key4 = 'vzFw4qYu-CcieFlcaznL-43CIA';
+
+// DOM Cache
+const $generateBtn = document.getElementById('search-btn') as HTMLButtonElement;
+const $inputField = document.getElementById('user-input') as HTMLInputElement;
+
+// Generate Listener
+document.addEventListener('DOMContentLoaded', () => {
+  $generateBtn?.addEventListener('click', generateFlashcard);
+});
+
+// Clicks
+document.addEventListener('click', (event) => {
+  const target = event.target as HTMLElement;
+
+  if (target.classList.contains('edit-btn')) handleEdit(target);
+  if (target.classList.contains('save-btn')) handleSave(target);
+  if (target.classList.contains('add-btn')) handleAddToDeck(target);
+});
```

Comment:
This works, but since you want this to do 3 different things for each different element, I probably would have just added the click event listeners to each button individually